### PR TITLE
Getting request total off of Content-Length header rather than event to bypass chrome bug

### PR DESCRIPTION
**Pull request by netanel-haber**
https://bugs.chromium.org/p/chromium/issues/detail?id=463622

### CODE REVIEW

**Comment by netanel-haber** [comment]
Hey :) I'm the original requester of https://github.com/squidfunk/mkdocs-material/issues/6087

File: src/templates/assets/javascripts/browser/request/index.ts [hunk 1]
```diff
@@ -85,7 +85,12 @@ export function request(
     // Handle download progress
     if (typeof options?.progress$ !== "undefined") {
       req.addEventListener("progress", event => {
-        options.progress$!.next((event.loaded / event.total) * 100)
+        if(event.lengthComputable){
```
Comment (squidfunk) [inline src/templates/assets/javascripts/browser/request/index.ts:88]: Please add whitespace around the braces ☺️

**Review by squidfunk — CHANGES_REQUESTED** [review]
Thanks, LGTM! Minor cosmetics.

**Comment by netanel-haber** [comment]
> Thanks, LGTM! Minor cosmetics.

Hey - just making sure - will this behavior be tested somehow? Is there a simple way for me to test it myself locally?

**Comment by squidfunk** [comment]
Well, I'm not sure _how_ we should test this automatically, since it's a browser bug. We need to rely on manual testing and best effort. I read your PR as "it works".

**Comment by netanel-haber** [comment]
> Well, I'm not sure _how_ we should test this automatically, since it's a browser bug. We need to rely on manual testing and best effort. I read your PR as "it works".

Loud and clear. Do you happen to have some recommended setup for local manual debugging/testing with a dummy website?

**Comment by squidfunk** [comment]
Not really, I think, we currently have no automated regression tests, and testing side effects like this is particularly challenging.

File: src/templates/assets/javascripts/browser/request/index.ts (hunk 1 above)
Comment (netanel-haber) [reply src/templates/assets/javascripts/browser/request/index.ts]: Done.

**Comment by netanel-haber** [comment]
Done checking, seems to work.
For what it's worth, what I did is the following, off the top of my head:

**Setting up the environment**:
I ran `npm run build`, then searched the minified js for the "progress" listener, then I swapped the augmented implementation in the minified js I have in my work Python docs project (after running `mkdocs build` locally). 

**Testing**:
* `If` branch (`event.lengthComputable===true`):
   Confirmed the loader still works in a non-buggy browser (firefox), in which I confirmed beforehand that `event.lengthComputable` is indeed `true`, so that tests the first branch.
* `else` branch:
   Checked in chrome and verified that the loader actually works, which was the initial goal of the PR.
  
Seems wonky and brittle to me, so if you do happen to have tips for the next PR, they would be much appreciated 😄

**Comment by squidfunk** [comment]
Thanks for testing! In order to test something like this automatically, we'd need to set up automated regression tests. We'll be working on setting up a test infrastructure next year, so we'll consider testing the frontend components as well.

**Comment by netanel-haber** [comment]
Actually, I was asking about an easier manual testing flow. Meaning, given a change/commit in the code, how do you usually pull up the changes live with some dummy website?

**Comment by squidfunk** [comment]
You can use our own documentation or any other site. I usually use `pip install -e <path-to-project>` to test any site against a specific commit.

**Pull request merged**